gemhardware/src/common/GEMHwDevice.cc

```diff
@@ -231,42 +91,51 @@ void gem::hw::GEMHwDevice::setup(std::string const& deviceName)
   m_ipBusErrs.Timeout       = 0;
   m_ipBusErrs.ControlHubErr = 0;
 
+  // uhal setup
   setLogLevelTo(uhal::Error());
-}
 
-uhal::HwInterface& gem::hw::GEMHwDevice::getGEMHwInterface() const
-{
-  if (p_gemHW == NULL) {
-    std::string msg = "Trying to access hardware before connecting!";
-    CMSGEMOS_ERROR("GEMHwDevice::" << msg);
-    XCEPT_RAISE(gem::hw::exception::UninitializedDevice, msg);
-  } else {
-    uhal::HwInterface& hw = static_cast<uhal::HwInterface&>(*p_gemHW);
-    return hw;
-  }
-  //have to fix the return value for failed access, better to return a pointer?
+  // xhal setup
+  connectRPC();
+  setLogLevel(1/*FIXME add enums to xhal, e.g., xhal::WARN*/);
 }
 
+// uhal::HwInterface& gem::hw::GEMHwDevice::getGEMHwInterface() const
+// {
+//   if (p_gemHW == NULL) {
+//     std::string msg = "Trying to access hardware before connecting!";
+//     CMSGEMOS_ERROR("GEMHwDevice::" << msg);
+//     XCEPT_RAISE(gem::hw::exception::UninitializedDevice, msg);
+//   } else {
+//     uhal::HwInterface& hw = static_cast<uhal::HwInterface&>(*p_gemHW);
+//     return hw;
+//   }
+//   //have to fix the return value for failed access, better to return a pointer?
+// }
+
+///////////////////////////////////////////////////////////////////////////////////////
+//****************Methods implemented for convenience on uhal devices****************//
+///////////////////////////////////////////////////////////////////////////////////////
+
 uint32_t gem::hw::GEMHwDevice::readReg(std::string const& name)
 {
   gem::utils::LockGuard<gem::utils::Lock> guardedLock(m_hwLock);
-  uhal::HwInterface& hw = getGEMHwInterface();
+  // uhal::HwInterface& hw = getGEMHwInterface();
 
   unsigned retryCount = 0;
   uint32_t res = 0x0;
-  CMSGEMOS_DEBUG("GEMHwDevice::gem::hw::GEMHwDevice::readReg "  << name << std::endl
-        << "Path  "      << hw.getNode(name).getPath() << std::endl
-        << "Address 0x"  << std::hex << hw.getNode(name).getAddress() << std::dec << std::endl
-        << "Mask 0x"     << std::hex << hw.getNode(name).getMask()    << std::dec << std::endl
-        << "Permission " << hw.getNode(name).getPermission() << std::endl
-        << "Mode "       << hw.getNode(name).getMode() << std::endl
-        << "Size "       << hw.getNode(name).getSize() << std::endl
-        << std::endl);
+  CMSGEMOS_DEBUG("GEMHwDevice::readReg " << name << std::endl
+                 << "Path  "      << this->getNode(name).getPath()                << std::endl
```

Comment:
Is there a global function shadowing `this->getNode`? Else `this->` isn't needed